tests/ignite/engine/test_deterministic.py

```diff
@@ -893,3 +894,27 @@ def test_engine_no_data_asserts():
 
     with pytest.raises(ValueError, match=r"Deterministic engine does not support the option of data=None"):
         trainer.run(max_epochs=10, epoch_length=10)
+
+
+def test_state_dict():
+    engine = DeterministicEngine(lambda e, b: 1)
+    sd = engine.state_dict()
+    assert isinstance(sd, Mapping) and len(sd) == 4
+    assert "iteration" in sd and sd["iteration"] == 0
+    assert "max_epochs" in sd and sd["max_epochs"] is None
+    assert "epoch_length" in sd and sd["epoch_length"] is None
+    assert "rng_states" in sd and sd["rng_states"] is not None
+
+    def _test(state):
```

Comment:
I think above tests are enough. Let's remove this _test

Reply:
will do.